src/cdef.rs

```diff
@@ -55,9 +56,12 @@ wrap_fn_ptr!(pub unsafe extern "C" fn cdef(
     damping: c_int,
     edges: CdefEdgeFlags,
     bitdepth_max: c_int,
-    _dst: *const FFISafe<Rav1dPictureDataComponentOffset>,
-    _top: *const FFISafe<CdefTop>,
-    _bottom: *const FFISafe<CdefBottom>,
+    dst: *const FFISafe<Rav1dPictureDataComponent>,
+    dst_offset: usize,
```

Comment:
I'm wondering now, does `WithOffset<*const FFISafe<Rav1dPictureDataComponent>` optimize okay?  I think, as it was for `bottom`, that the pointer/reference indirection is throwing off the optimization, and I would hope that `WithOffset<T>` and `T, usize` can still be optimized the same.

Reply:
🤯 yep that seems to work - changed to that and set `repr(C)` in `WithOffset` and got the nice stack usage.